Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs

```diff
@@ -77,6 +90,63 @@ public void InitializeWithMatcherShouldSetCorrespondingProperty()
             Assert.Equal(matcher, sut.Matcher);
         }
 
+
+        [Fact]
+        public void InitializeWithSingleNullRequestArgumentShouldThrowArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null));
+        }
+
+        [Fact]
+        public void InitializeWithNullRequestShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null, new FalseRequestSpecification()));
+            Assert.Equal("request", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithRequestAndNullMatcherShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                  new FreezeOnMatchCustomization(new object(), null));
+            Assert.Equal("matcher", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithSingleRequestShouldSetCorrespondingProperty()
+        {
+            var request = new object();
+            var sut = new FreezeOnMatchCustomization(request);
+            Assert.Same(request, sut.Request);
+        }
+
+        [Fact]
+        public void InitializeWithObjectRequestShouldSetTrueRequestSpecificationMatcher()
```

Comment:
> This code is redundant and can be simplified to the EqualRequestSpecification only, but you don't want to simplify it because of this test which is a pure structural inspection.

I wouldn't say so. This PR solves the original issue and is well covered with tests which enables further refactoring. Since it's one year old PR, at this particular moment I have no idea what's going on inside because I simply do not remember ;)

I (probably) haven't dig into details so much to notice that the mentioned `if-else` statement can be simplified. If that's true and the only 'blocker' is those structural test, I'm fine with killing it ;)